drivers/regex/mlx5/mlx5_regex.h

```diff
@@ -35,6 +35,8 @@ struct mlx5_regex_priv {
 	uint16_t nb_queues; /* Number of queues. */
 	struct mlx5_regex_qp *qps; /* Pointer to the qp array. */
 	uint16_t nb_max_matches; /* Max number of matches. */
+	//ORI check, where best to add this mode? :
+	enum mlx5_rxp_program_mode prog_mode;
```

Comment:
currenlty just assume it is always shared, it will be added using device parameters